frontend/app/routes/create.$type.tsx

```diff
@@ -75,35 +101,53 @@ export default function Create() {
     defaultConfig,
     ilpayUrl,
     scriptInitUrl,
-    contentOnlyParam
+    walletAddress,
+    contentOnlyParam,
+    isGrantAccepted,
+    grantResponse,
+    isGrantResponse,
+    lastAction
   } = useLoaderData<typeof loader>()
   const response = useActionData<typeof action>()
   const { state } = useNavigation()
+  const location = useLocation()
   const isSubmitting = state === 'submitting'
   const contentOnly = contentOnlyParam != null
 
   const [openWidget, setOpenWidget] = useState(false)
   const [toolConfig, setToolConfig] = useState<ElementConfigType>(defaultConfig)
   const [fullConfig, setFullConfig] =
     useState<Record<string, ElementConfigType>>()
-  const [modalOpen, setModalOpen] = useState(false)
-  const [importModalOpen, setImportModalOpen] = useState(false)
-  const [newVersionModalOpen, setNewVersionModalOpen] = useState(false)
-  const [infoModalOpen, setInfoModalOpen] = useState(false)
-  const [confirmModalOpen, setConfirmModalOpen] = useState(false)
+  const [modalOpen, setModalOpen] = useState<ModalType | undefined>()
+
   const [selectedVersion, setSelectedVersion] = useState('default')
   const [versionOptions, setVersionOptions] = useState<SelectOption[]>([
     { label: 'Default', value: 'default' }
   ])
   const [configUpdateTrigger, setConfigUpdateTrigger] = useState(0)
 
-  const wa = (toolConfig?.walletAddress || '')
-    .replace('$', '')
-    .replace('https://', '')
+  const wa = toWalletAddressUrl(toolConfig?.walletAddress || '')
   const scriptToDisplay = `<script id="wmt-init-script" type="module" src="${scriptInitUrl}init.js?wa=${wa}&tag=[version]&types=[elements]"></script>`
   const submitForm = useSubmit()
 
-  const onConfirm = () => {
+  const getFormData = (
+    configArray: Record<string, ElementConfigType>,
+    intent: string,
+    version: string
+  ) => {
+    const formData = new FormData()
+    formData.append('intent', intent)
+    formData.append('version', version)
+    formData.append('fullconfig', JSON.stringify(configArray))
+    const defaultSet = configArray.default as unknown as Record<string, string>
+    Object.keys(defaultSet).map((key) => {
+      formData.append(key, defaultSet[key])
+    })
```

Comment:
Recommend setting required-ish fields later, in case some key from `defaultSet` overrides intent/version/fullconfig in future. And using `formData.set` instead of append so there's a single key (otherwise might need to use `getAll` in backend).

```suggestion
    const defaultSet = configArray.default as unknown as Record<string, string>
    Object.keys(defaultSet).map((key) => {
      formData.set(key, defaultSet[key])
    })
    formData.set('intent', intent)
    formData.set('version', version)
    formData.set('fullconfig', JSON.stringify(configArray))
```

Reply:
there's no overlap regarding the data, but easy enough to change the order, so it's done